pxr/usd/sdf/path.cpp

```diff
@@ -1787,48 +1789,86 @@ SdfPath::MakeRelativePath(const SdfPath & anchor) const
     return result;
 }
 
+static inline bool _IsValidIdentifier(const std::string& name)
+{
+    // empty strings are not valid identifiers
+    if (name.empty())
+    {
+        return false;
+    }
+
+    TfUtf8CodePointView view {name};
+    bool first = true;
+    for (const uint32_t codePoint : view)
+    {
+        bool result = first ? 
+            ((codePoint == SDF_UNDERSCORE_CODE_POINT) ||
+                TfIsUtf8CodePointXidStart(codePoint))
+            : TfIsUtf8CodePointXidContinue(codePoint);
+
+        if (!result)
+        {
+            return false;
+        }
+
+        first = false;
+    }
+
+    return true;
+}
+
 static inline bool _IsValidIdentifier(TfToken const &name)
 {
-    return TfIsValidIdentifier(name.GetString());
+    return _IsValidIdentifier(name.GetString());
 }
 
 bool
 SdfPath::IsValidIdentifier(const std::string &name)
 {
-    return TfIsValidIdentifier(name);
-}
-
-// We use our own _IsAlpha and _IsAlnum here for two reasons.  One, we want to
-// ensure that they follow C/Python identifier rules and are not subject to
-// various locale differences.  And two, since we are not consulting a locale,
-// it is faster.
-static constexpr bool _IsAlpha(int x) {
-    return ('a' <= (x|32)) && ((x|32) <= 'z');
-}
-static constexpr bool _IsAlnum(int x) {
-    return _IsAlpha(x) || (('0' <= x) && (x <= '9'));
+    return _IsValidIdentifier(name);
 }
 
 bool
 SdfPath::IsValidNamespacedIdentifier(const std::string &name)
 {
+    // empty strings are not valid identifiers
+    if (name.empty())
+    {
+        return false;
+    }
+
     // A valid C/Python identifier except we also allow the namespace delimiter
     // and if we tokenize on that delimiter then all tokens are valid C/Python
-    // identifiers.  That means following a delimiter there must be an '_' or
-    // alphabetic character.
-    constexpr char delim = SDF_PATH_NS_DELIMITER_CHAR;
-    for (char const *p = name.c_str(); *p; ++p) {
-        if (!_IsAlpha(*p) && *p != '_') {
+    // identifiers.
+    std::string_view remainder {name};
+    while (!remainder.empty()) {
+        const auto index = remainder.find(':');
+        
+        // can't start with ':'
+        if (index == 0) {
             return false;
         }
-        for (++p; _IsAlnum(*p) ||*p == '_'; ++p) {
-            /* consume identifier */
+
+        // can't end with ':'
+        if (index == remainder.size() - 1) {
+            return false;
         }
-        if (*p != delim) {
-            return !*p;
+
+        // substring must be a valid identifier
+        if (!SdfPath::IsValidIdentifier(std::string(remainder.substr(0, index)))) {
```

Comment:
In lieu of creating a `std::string`, can we just make the internal `_IsValidIdentifier` (which `SdfPath::IsValidIdentifier` calls) take a `string_view`?